### DISCUSSION TITLE
Possible to pass heuristics to CBC (or other MILP solvers)?

### DISCUSSION THREAD

**Discussion by carolinetjes** [casadi/casadi#3158] [Q&A]
Dear all,

To speed up performance I'd like to **pass heuristics** to Mixed Integer Linear Program (MILP) solvers as they are called via Casadi - is this possible? In particular, the question is relevant for CBC (see [cbcheuristics)](https://www.coin-or.org/Cbc/ch03s02.html ), but would also be interested in hearing about other solvers if that makes a difference for the answer.

I read the Casadi callbacks docs but remained inconclusive. 

A yes/no answer would already be great - some hints or a small working example if you happen to have one even more amazing.  Thanks!

**Comment by jaeandersson** [comment]
I think the answer is no, but I haven't touch that code in a long time. @jgillis, do you know? I don't enough about it to say how hard it would be to add/interface.

**Reply by carolinetjes** [reply]
Many thanks for the swift initial reply!

**Comment by jgillis** [comment]
Hi Caroline,

Do you mean custom heuristics besides the built-in ones accessible through options like `feasibilityPump` and `greedyHeuristic` (some listed here https://www.gams.com/latest/docs/S_CBC.html)?
Then, no, and not in any of our interfaces.

Me and Joel hardly ever use MIP nowadays; the CBC interface was part of contracted work.

**Reply by carolinetjes** [reply]
I did indeed mean custom heuristics. Good to know it is not an option; we will pursue a different route.

I'm thinking we'll switch to HiGHS and pass merely a feasible solution. That should be much easier than a whole heuristic. If we run into anything there, I will start a new thread. For now, thanks 🙋🏻‍♀️
